client/src/components/layout/Vote.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from 'react'
+import { withRouter } from 'react-router'
+
+const Vote = props => {
+
+  const reviewId = props.reviewId
+  const [voteCount, setVoteCount] = useState({
+    upVote: 0,
+    downVote: 0
+  })
+
+  const getVoteCount = async () => {
+    try {
+      const response = await fetch(`/api/v1/votes/${reviewId}`)    
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
+      }
+      const body = await response.json()
+      setVoteCount(body.totalCount)
+    } catch (error) {
+      return console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  useEffect(() => {
+    getVoteCount()
+  }, [])
+
+  const addVote = async (voteValue) => {
+    try {
+      const response = await fetch(`/api/v1/votes/${reviewId}`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify({
+          voteValue: voteValue
+        }) 
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+          alert(body.message)  
+        }
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
+      }
+      const body = await response.json()
+      setVoteCount(body.totalCount)
+    } catch (error) {
+      return console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  const upClickHandler = async () => {
+    await addVote(1)
+  }
+
+  const downClickHandler = async () => {
+    await addVote(-1)
+  }
+
+  const deleteVote = async () => {
+    try {
+      const response = await fetch('/api/v1/votes', {
+        method: "DELETE",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify({ reviewId }) 
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+          alert(body.message)  
+        }
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
+      }
+      const body = await response.json()
+      setVoteCount(body.totalCount)
+    } catch (error) {
+      return console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  let upButton
+  let downButton
+  let deleteButton
+  let signInMessage = <p>Sign in to vote</p>
+  if (props.user) {
+    signInMessage = null
+    upButton = <button className="button" onClick={upClickHandler}>Upvote</button>
+    downButton = <button className="button" onClick={downClickHandler}>Downvote</button>
+    deleteButton = <button className="button" onClick={deleteVote}>Delete my vote</button>
```

Comment:
Having a separate delete option is a little odd - typically we would expect that clicking a vote button once saves my option, and clicking it a second time will undo that option, similar to Reddit for example. The behavior couples nicely with ensuring a user can only make a vote selection once, either up or down. Selecting a different option changes the saved vote, and selecting the same option will undo/remove/delete the vote